pcdsdevices/lasers/tuttifrutti.py

```diff
@@ -0,0 +1,146 @@
+"""
+Classes for the Tuttifrutti diagnostic stack designed for the MODS TILEs.
+"""
+
+from ophyd import Component as Cpt
+from ophyd import Device
+from ophyd.device import create_device_from_components
+
+from pcdsdevices.areadetector.detectors import PCDSAreaDetectorTyphos
+from pcdsdevices.lasers.qmini import QminiSpectrometer
+from pcdsdevices.lasers.ek9000 import El3174AiCh
+from pcdsdevices.lasers.elliptec import Ell6
+
+
+def TuttiFruttiCls(prefix, name, nf=False, ff=False, spec=False, pm=False,
+                   diode=False, em=False, qc=False, pd=False, wfs=False,
+                   ell=False, misc=[]):
+    """
+    Generate a TuttiFrutti class. See TuttiFrutti function for more details.
+    """
+    cpts = {}
+    if nf:
+        cpt = Cpt(PCDSAreaDetectorTyphos, '_NF1:', kind='normal')
+        cpts['nf_camera'] = cpt
+    if nf:
+        cpt = Cpt(PCDSAreaDetectorTyphos, '_FF1:', kind='normal')
+        cpts['ff_camera'] = cpt
+    if spec:
+        cpt = Cpt(QminiSpectrometer, '_SP1', kind='normal')
+        cpts['spectrometer'] = cpt
+    if pm:
+        cpt = Cpt(El3174AiCh, '_PM1', kind='normal')
+        cpts['power_meter'] = cpt
+    if diode:
+        raise(NotImplemented, "Diode is not yet implemented")
+    if em:
+        raise(NotImplemented, "Energy meter is not yet implemented")
+    if qc:
+        raise(NotImplemented, "Quad cell is not yet implemented")
+    if pd:
+        raise(NotImplemented, "Pulse duration is not yet implemented")
+    if wfs:
+        raise(NotImplemented, "Wavefront sensor is not yet implemented")
+    if ell:
+        cpt = Cpt(Ell6, '_SL1:ELL:M1', kind='normal')
+        cpts['slider'] = cpt
+    if misc:  # This feels kind of hacky, but also kind of cool.
+        for cptname, cpt in misc.items():
+            cpts[cptname] = cpt
```

Comment:
Do you expect this to get a lot of use? Just curious. It seems fine, but I thought the Tuttifrutti were pretty modular so having a misc surprises me.

Reply:
I hope not. It just seemed like a nice convenience for dealing with off-normal deployments. If we start instantiating TuttiFruttis outside of the MODS, like they plan to in XPP, I can't guarantee that the components will be the same.

Reply:
Fair enough!